client/galaxy/scripts/components/Workflow/Editor/Node.vue

```diff
@@ -80,6 +113,18 @@ export default {
         onClone() {
             this.node.clone();
         },
+        onCreate(toolId, event) {
```

Comment:
This is duplication of `onInsertTool` in `Index.vue`. Please emit an event `this.$emit("onInsertTool", ...)` instead and handle the request in `Index.vue` for now.

Reply:
They are similar but in `onCreate(toolId, event) ` call, we only have `tool_id` and do not have `tool_name`. We get it by the service call `getModule` and then using the tool's name and id, we create tool's node (toolbox). We can emit an event from Node.vue and then still need to have a separate method in Index.vue

Reply:
Additionally, we hide the popover in Node.vue `this.popoverShow = false` after creating the toolbox

Reply:
Then we we can parse a custom event and handle it in the parent. The `popoverShow` can be set to `false` before emitting the event.